File: ch36p906XML/p741p758XML/MyPeopleParser.m


#import "MyPeopleParser.h"
#import "MyPersonParser.h"

@implementation MyPeopleParser


- (void)parser:(NSXMLParser *)parser didStartElement:(NSString *)elementName namespaceURI:(NSString *)namespaceURI qualifiedName:(NSString *)qualifiedName attributes:(NSDictionary *)attributeDict
{
    if ([elementName isEqualToString: @"people"])
        self.people = [NSMutableArray array];
    if ([elementName isEqualToString: @"person"]) {
        [self makeChild:[MyPersonParser class] elementName:elementName parser:parser];
    }
}

- (void) finishedChild: (NSString*) s {
    [self.people addObject: [(MyPersonParser*)self.child person]];
}


@end

File: iOS8bookExamples/bk2ch23p818xml/ch36p1070xml/MyPeopleParser.swift


import UIKit

class MyPeopleParser : MyXMLParserDelegate {
    
    var people = [Person]()
    
    func parser(parser: NSXMLParser, didStartElement elementName: String,
        namespaceURI: String?, qualifiedName qName: String?,
        attributes attributeDict: [NSObject : AnyObject]!) {
            if elementName == "person" {
                self.makeChild(MyPersonParser.self, elementName: elementName, parser: parser)
            }
    }
    override func finishedChild(s: String) {
        self.people.append((self.child as! MyPersonParser).person)
    }
    
}
